### DISCUSSION TITLE
Release 3.6.0

### DISCUSSION THREAD

**Discussion by blapie** [shibatch/sleef#498] [General]
Hello,
 
We are aware that users have been waiting for release 3.6.0 to come out for a while now. Due to a limited access to actual machines we have been focusing on setting up sufficient test coverage based on emulation, fixing a few outstanding bugs and documentation.
It is likely the upcoming release will officially support a subset of the OS-es and compilers it used to claim support for, but should just work on most of them. More bug fixes will be available in the following minor releases.

We are now expecting to tag release 3.6.0 in a matter of weeks.

**Comment by blapie** [comment]
Hi!

As of last week we did not accept any more contributions for this release. Validation and out-of-the-box testing have been successfully performed. Therefore, I will tag the release 3.6 some time this week.

We are aware that there are still limitations in the range of features that are currently tested, and we made sure to document them on the README and issue #481. More releases should follow this year for which we plan to solve a few existing bugs, add more features and extend test coverage.

As always we encourage everyone to contribute, be it on the library, build system, or CI side - but also on code reviews. Many of the features and CI available in this release are attributed to external contributors. The library benefits greatly from these contributions, and we are very grateful for that!

**Comment by blapie** [comment]
Hi,
I just created the long overdue release 3.6 today. Please find a note for the newly tagged release [here](https://github.com/shibatch/sleef/releases/tag/3.6).
More frequent releases should follow this one.

**Comment by blapie** [comment]
@adamjstewart Are you the right person to contact for updating Spack with the new version of SLEEF?

@cdluminate Are you still responsible for packaging for Debian and Ubuntu PPA?

**Reply by adamjstewart** [reply]
I don't usually update the Spack package unless I need a newer version (depends on PyTorch). If anyone else is interested in helping to maintain our Spack recipe I can review and merge the PR. It should be as simple as a 1-line PR.

**Reply by blapie** [reply]
Right, I understand. I was suspecting some interactions with PyTorch.
Will send a PR when I have a minute. Thanks for the update!

**Reply by cdluminate** [reply]
> @cdluminate Are you still responsible for packaging for Debian and Ubuntu PPA?

Yes for Debian. Ubuntu PPA is never covered by me.

**Reply by blapie** [reply]
@cdluminate Ok, thank you for confirming! Will update the info on the website accordingly.

**Reply by blapie** [reply]
@cdluminate I noticed that the test pipeline failed to compile on 3.6 https://ci.debian.net/packages/s/sleef/unstable/amd64/43343207/
That is likely due to a change in location for the test file. Can I help modify the test so it can run normally?

**Reply by cdluminate** [reply]
That error is fine for now. I'll fix it for the next upload. sleef 3.6.0 is currently in experimental. I do not plan to make a transition to unstable because pytorch 2.1.2 does not build against sleef 3.6.0 on amd64.

**Comment by xuhancn** [comment]
@blapie When you plan to resume the PR merge?

**Reply by blapie** [reply]
Hi! Unfortunately we have limited bandwidth and need to prioritise maintenance work. We needed to get the release out of the way first and therefore we've had limited time to review the more significant changes. We will be in touch as soon as possible for your PR.

**Reply by xuhancn** [reply]
Thanks for reply.

**Comment by xuhancn** [comment]
@blapie I suggest to release 3.6.1, since we fix a lot of bugs.

**Reply by blapie** [reply]
It is planned now. https://github.com/shibatch/sleef/discussions/538

**Comment by joanaxcruz** [comment]
Thank you for your suggestion Xu Han. We currently have no plans for a patch release this quarter, but we'll certainly give it some consideration. If we are able to, then we'll confirm the scope and timeframe as soon as we can.
